content/rest/guides/traversing-with-pagination.md

```diff
@@ -84,6 +84,23 @@ and more importantly, `rel="prev"` lets you know the page number of the previous
 page. Using this information, you could construct some UI that lets users jump
 between the first, previous, next, or last list of results in an API call.
 
+### Example of Python Iteration over pages
+
+While getting data from GitHub and the amount of pages is unknown, the `reqeusts` can
+be `while` until the `next` will end, and we'll reach the last page and only `prev` and `first` will
+show in the headers.
+
+```python
+import requests
+
+url = "https://api.github.com/XXXX?simple=yes&per_page=100&page=1"
+res=requests.get(url,headers={"Authorization": git_token})
+repos=res.json()
+while 'next' in res.links.keys():
+  res=requests.get(res.links['next']['url'],headers={"Authorization": git_token})
+  repos.extend(res.json())
+```
+
```

Comment:
````suggestion
### Example: Iterating over pages with a script

Instead of requesting each page of results individually, you can write a script to iterate over all of the pages. The following example uses the Python `requests` library. The `while` loop continues to make requests as long is there is a next page of results.

```python
import requests

token = os.environ.get("YOUR_TOKEN")

url = "https://api.github.com/search/code?q=addClass+user:mozilla&per_page=100&page=1"
result = requests.get(url, headers={"Authorization": token})
repos = result.json()
while 'next' in result.links.keys():
  result = requests.get(result.links['next']['url'], headers={"Authorization": token})
  repos.extend(result.json())
```

````